website/citations/views.py

```diff
@@ -183,7 +196,87 @@ def citation_list(self):
         def _citation_list(auth, node_addon, list_id=None, **kwargs):
             """ Returns a list of citations
             """
+
             show = request.args.get('view', 'all')
-            return Provider().citation_list(node_addon, auth.user, list_id, show)
+            if not waffle.flag_is_active(request, features.ENABLE_GV):
+                return Provider().citation_list(node_addon, auth.user, list_id, show)
+            else:
+                response = dict(contents=[])
+                project_ulr = settings.DOMAIN + request.view_args.get('pid')
+                project = Node.objects.filter(guids___id__in=[request.view_args.get('pid')]).first()
+                resource_references_response = get_gv_result(
+                    endpoint_url=RESOURCE_LIST_ENDPOINT,
+                    requesting_user=auth.user,
+                    requested_resource=project,
+                    params={'filter[resource_uri]': project_ulr},
+                )
+                configured_storage_addons_url = resource_references_response.get_related_link(
+                    'configured_storage_addons')
+                addons_url_list = get_gv_result_json(
+                    endpoint_url=configured_storage_addons_url,
+                    requesting_user=auth.user,
+                    requested_resource=project,
+                    request_method='GET',
+                    params={}
+                )
```

Comment:
you can use `get_addon` call for that.
It will return configured addon with external service included, whom which you can get `wb_key` and filter only requested addons